### PR TITLE
Cleaned up formik components

### PR DESCRIPTION
**Pull request by Nuzair46**

### CODE REVIEW

**Comment by amaldinesh7** [comment]
@Nuzair46 Please do add the issue number which this PR closes. Update the same in all your PRs.

**Comment by Nuzair46** [comment]
@amaldinesh7 _a, Please review. Thanks

**Comment by amaldinesh7** [comment]
Why is the circle CI config changes reflected in this this PR? did you merge master instead of rebasing? Please do correct it. @Nuzair46

File: lib/components/formik/BlockNavigation.js [hunk 1]
```diff
@@ -4,11 +4,12 @@ import { useFormikContext } from "formik";
 import { useNavPrompt } from "react-router-nav-prompt";
 import PropTypes from "prop-types";
 
-function NavigationHandler({ isDirty = false }) {
+const NavigationHandler = ({ isDirty = false }) => {
   const formikContext = useFormikContext();
   const shouldBlock =
-    isDirty || (formikContext ? Boolean(formikContext.dirty) : false);
+    isDirty || (formikContext && Boolean(formikContext.dirty));
```
Comment (amaldinesh7) [inline lib/components/formik/BlockNavigation.js:10]: `shouldBlock` should be a boolean. if the formikContext isn't true, what will be its value?

File: lib/components/formik/Select.js [hunk 2]
```diff
@@ -8,8 +8,8 @@ import { either, isNil, isEmpty } from "ramda";
 
 const SelectField = React.forwardRef((props, ref) => {
```
Comment (amaldinesh7) [inline lib/components/formik/Select.js:9]: Please import forwardRef and use it.

**Review by amaldinesh7 — CHANGES_REQUESTED** [review]
@Nuzair46 _a Please check the comments.

**Pull request closed without merging**